nym-network-monitor/src/main.rs

```diff
@@ -0,0 +1,204 @@
+use crate::http::HttpServer;
+use accounting::submit_metrics;
+use anyhow::Result;
+use clap::Parser;
+use log::{info, warn};
+use nym_crypto::asymmetric::ed25519::PrivateKey;
+use nym_network_defaults::setup_env;
+use nym_network_defaults::var_names::NYM_API;
+use nym_sdk::mixnet::{self, MixnetClient};
+use nym_topology::{HardcodedTopologyProvider, NymTopology};
+use std::fs::File;
+use std::io::Write;
+use std::sync::LazyLock;
+use std::{
+    collections::VecDeque,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    str::FromStr,
+    sync::Arc,
+};
+use tokio::sync::OnceCell;
+use tokio::{signal::ctrl_c, sync::RwLock};
+use tokio_util::sync::CancellationToken;
+
+static NYM_API_URL: LazyLock<String> = LazyLock::new(|| {
+    std::env::var(NYM_API).unwrap_or_else(|_| panic!("{} env var not set", NYM_API))
+});
+
+static MIXNET_TIMEOUT: OnceCell<u64> = OnceCell::const_new();
+static TOPOLOGY: OnceCell<NymTopology> = OnceCell::const_new();
```

Comment:
don't you need to periodically change it as new nodes bond/old ones unbond?

Reply:
The idea is that monitor is a short lived process (shorter then the epoch) that bootstraps its topology every run